### PR TITLE
Feature/refactor basket api

### LINKED ISSUE #175: Send all item data in basket json object
```
{
  "basketID": "3fa85f64-5717-4562-b3fc-2c963f66afa6",
  "items": {
    "itemid": 1,
    "itemid": 0,
    "itemid": 0
  },
  "locked": true,
  "paid": true,
  "redeemRequest": "string",
  "redeemed": true,
  "rewardItems": [
    "string"
  ],
  "value": 0
}
```
Basket json objects do not contain the item data, which makes the basket api nasty to work with.

### PR DESCRIPTION
**Pull request by this-kramer**
## Reason for this PR:

- Refactor basket api to send all item data with basket instead of only the item id (for shopping items and reward items)
- Closes #175 

### New basket json
```
{
  "basketID": "3fa85f64-5717-4562-b3fc-2c963f66afa6",
  "basketItems": [
    {
      "count": 0,
      "id": "string",
      "price": 0,
      "title": "string"
    }
  ],
  "locked": true,
  "paid": true,
  "redeemRequest": "string",
  "redeemed": true,
  "rewardItems": [
    {
      "id": "string",
      "title": "string"
    }
  ],
  "value": 0
}
```

## Checklist:

Select whatever applies:

- [x] Test written
- [x] Code Properly Formatted
- [x] Documented

### CODE REVIEW

File: services/basket/src/main/java/org/cryptimeleon/incentive/services/basket/model/BasketItemNew.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import io.swagger.annotations.ApiModelProperty;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.cryptimeleon.incentive.services.basket.storage.ItemEntity;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class BasketItemNew {
```
Comment (pschuermann97) [inline services/basket/src/main/java/org/cryptimeleon/incentive/services/basket/model/BasketItemNew.java:12]: why the weird name?

File: services/bootstrap/src/main/java/org/cryptimeleon/incentive/services/bootstrap/DefaultBootstrapData.java [hunk 2]
```diff
@@ -19,32 +19,32 @@
 
 public class DefaultBootstrapData {
 
-    private static final BasketItemDto[] basketItems = {
-            new BasketItemDto(
+    private static final ItemDto[] basketItems = {
+            new ItemDto(
                     "3941288190038",
                     "Sweetened Hazelnut Spread",
```
Comment (pschuermann97) [inline services/bootstrap/src/main/java/org/cryptimeleon/incentive/services/bootstrap/DefaultBootstrapData.java:25]: Debranding done right :)

**Review by pschuermann97 — APPROVED** [review]
approved

File: services/basket/src/main/java/org/cryptimeleon/incentive/services/basket/model/BasketItemNew.java (hunk 1 above)
Comment (this-kramer) [reply services/basket/src/main/java/org/cryptimeleon/incentive/services/basket/model/BasketItemNew.java]: Good catch, this name was meant to exists only temporarily

File: services/bootstrap/src/main/java/org/cryptimeleon/incentive/services/bootstrap/DefaultBootstrapData.java (hunk 2 above)
Comment (this-kramer) [reply services/bootstrap/src/main/java/org/cryptimeleon/incentive/services/bootstrap/DefaultBootstrapData.java:25]: :D

**Review by pschuermann97 — APPROVED** [review]
approved

File: services/basket/src/main/java/org/cryptimeleon/incentive/services/basket/api/Basket.java [hunk 3]
```diff
@@ -20,49 +20,46 @@ public class Basket {
...
+     * The first redeem request that the user sent for this basket is stored.
+     * This allows for the user to safely retry redeeming in case of connectivity problems
+     * without allowing for easy double-spending attacks.
+     */
     @ApiModelProperty(value = "${basketModel.redeemRequest}")
     private String redeemRequest;
 
     @ApiModelProperty(value = "${basketModel.value}")
-    private long value; // value must be set manually for serialization
+    private long value;
```
[the remark this answers is not in the document]
Comment (this-kramer) [reply services/basket/src/main/java/org/cryptimeleon/incentive/services/basket/api/Basket.java:57]: What do you mean? It's the value/price of the basket 🤔

**Pull request merged**